### DISCUSSION TITLE
Regression: Gitlab Sub Module tag discovery issue

### DISCUSSION THREAD

**Discussion by bassforce86** [renovatebot/renovate#30913] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab v17.3.0, Renovate: 38.40.1

### Please tell us more about your question or problem

When fetching version updates for GitLab sub-modules, Renovate appears to be taking the latest tag from the main repository, instead of filtering sub modules by the tag prefix names.

e.g:
```yaml
 - repo location: gitlab.com/company/lib
    version: v1.39.0
 - sub-module location: gitlab.com/company/lib/log
    version: log/v0.1.2 
```

This _should_ produce (presuming both require an update) 
  - update company/lib from: v1.34.0 to: v1.39.0
  - update company/lib/log from: v0.1.1 to: v0.1.2

But instead we get 
  - update company/lib from: v1.34.0 to: v1.39.0 ✅ 
  - update company/lib/log from: v0.1.1 to: v1.39.0 🚫 

### Logs (if relevant)

<details><summary>Logs</summary>

```

INFO: Repository started (repository=company/project)
       "renovateVersion": "38.40.1"
 INFO: Dependency extraction complete (repository=company/project, baseBranch=main)
       "stats": {
         "managers": {"gomod": {"fileCount": 1, "depCount": 47}},
         "total": {"fileCount": 1, "depCount": 47}
       }
 INFO: Branch updated (repository=company/project, branch=renovate/all)
       "commitSha": "35d8cbc9b3786562afab88f81ec6191a3a96137f"
 WARN: No github.com token has been configured. Skipping release notes retrieval (repository=company/project, branch=renovate/all)
       "manager": "gomod",
       "packageName": "go",
       "sourceUrl": "https://github.com/golang/go"
 INFO: PR updated (repository=company/project, branch=renovate/all)
       "pr": 10,
       "prTitle": "fix(deps): update all dependencies"
 WARN: artifactErrors (repository=company/project, branch=renovate/all)
       "artifactErrors": [
         {
           "lockFile": "go.sum",
           "stderr": "Command failed: go get -d -t ./...\ngo: -d flag is deprecated. -d=true is a no-op\ngo: gitlab.com/company/lib/log@v1.39.0: reading gitlab.com/company/lib/log/go.mod at revision log/v1.39.0: unknown revision log/v1.39.0\n"
         }
       ]
 INFO: Repository finished (repository=company/project)
       "cloned": true,
       "durationMs": 37891
 INFO: Renovate was run at log level "info". Set LOG_LEVEL=debug in environment variables to see extended debug logs.

```

</details>

**Comment by bassforce86** [comment]
I have confirmed the regression by downgrading and pinning our renovate image to `38.18.7` which was the release associated with the original fix for this in [issue #28540](https://github.com/renovatebot/renovate/issues/28540) which correctly reported the version increments

**Reply by rarkins** [reply]
Which version of Renovate broke things?

**Reply by rarkins** [reply]
Do you mean 38.18.7 broke things while 38.18.6 works?

**Reply by bassforce86** [reply]
Oh, appologies I thought i'd added the current version in the openning post. I'm afraid i don't know exactly _when_ this broke, as we only run renovate periodically, but we first noticed issues running version `38.40.1`

**Reply by bassforce86** [reply]
reverting back to `38.18.7` fixed things for us

**Reply by rarkins** [reply]
OK, we will wait until you or another affected user is able to pinpoint which release/commit introduced the regression

**Reply by bassforce86** [reply]
Yep, not to worry - if I find any more info ill report back.
